src/subspace_store.c

```diff
@@ -66,7 +67,8 @@ subspace_store_internal_node_descendants(SubspaceStoreInternalNode *node, int in
 SubspaceStore *
 subspace_store_init(Hyperspace *space, MemoryContext mcxt, int16 max_items)
 {
-	MemoryContext old = MemoryContextSwitchTo(mcxt);
+	MemoryContext storemctx = AllocSetContextCreate(mcxt, "chunk store memory context", ALLOCSET_DEFAULT_SIZES);
```

Comment:
Why do we need to create a new memory context here now?

Reply:
Also where is this context deleted?

Reply:
This puts each chunk in its own child memory context. Deleted when the parent is deleted.

Reply:
It allows you to have a per-hypertable chunk cache. Not strictly needed without the chunk relid cache I implemented. It is deleted when the parent (hypertable cache memory context) is deleted. I think it makes sense but can revert if you feel strongly about it.